go/vt/wrangler/reparent.go

```diff
@@ -405,6 +405,18 @@ func (wr *Wrangler) plannedReparentShardLocked(ctx context.Context, ev *events.R
 		return fmt.Errorf("old master tablet %v DemoteMaster failed: %v", topoproto.TabletAliasString(shardInfo.MasterAlias), err)
 	}
 
+	// When using RBR and semi-sync, new master can take a long time to find GTID set to send to replicas.
+	// This causes slow writes when a master gets promoted. Full details in:
+	// https://github.com/vitessio/vitess/issues/4161
+	// Flushing binlogs before promoting a master fixes this problem.
+	wr.logger.Infof("flushing binary logs in target master %v", masterElectTabletAliasStr)
+	event.DispatchUpdate(ev, "flush binary logs")
+	err = wr.tmc.FlushBinaryLogs(ctx, masterElectTabletInfo.Tablet)
+	if err != nil {
+		wr.logger.Warningf("Could not flush binary logs in tablet: %v. New master could be slower to take writes", masterElectTabletAlias)
```

Comment:
IMHO we should drop the "New master could be slower to take writes" editorial from this log.

Though helpful I think it's also potentially misleading as the issue is (a) only relevant in semi-sync mode as discussed above and (b) is technically a problem with commits and depends on the max_semi_sync_wait_timeout setting.

Better to just log what didn't work and why and leave it up to the user to figure out what that means for their environment.